### DISCUSSION TITLE
Compliance policies | Compliance policy settings

### DISCUSSION THREAD

**Discussion by TituNL** [Micke-K/IntuneManagement#131] [Q&A]
Hi,

First of all i would like to thank the creators of the IntuneManagement app. It really makes me work more efficiently.
There is only 1 think that i miss in the app atm.

Is it possible to configure "[Compliance policies | Compliance policy settings](https://endpoint.microsoft.com/#view/Microsoft_Intune_DeviceSettings/DevicesComplianceMenu/~/policySettings)" with the IntuneManagement app?

**Graph code for it:**

`{
    "@odata.context": "https://graph.microsoft.com/beta/$metadata#deviceManagement/settings",
    "deviceComplianceCheckinThresholdDays": 30,
    "isScheduledActionEnabled": true,
    "secureByDefault": true,
    "enhancedJailBreak": false,
    "deviceInactivityBeforeRetirementInDay": 0,
    "derivedCredentialProvider": "notConfigured",
    "derivedCredentialUrl": null,
    "androidDeviceAdministratorEnrollmentEnabled": false,
    "ignoreDevicesForUnsupportedSettingsEnabled": false,
    "enableLogCollection": true,
    "enableAutopilotDiagnostics": true,
    "enableEnhancedTroubleshootingExperience": false,
    "enableDeviceGroupMembershipReport": false
}`

**Powershell version:**
`Import-Module Microsoft.Graph.DeviceManagement
$params = @{
	Settings = @{
		DeviceComplianceCheckinThresholdDays = 30
		IsScheduledActionEnabled = $true
		SecureByDefault = $true
		EnhancedJailBreak = $false
		DeviceInactivityBeforeRetirementInDay = 0
		DerivedCredentialProvider = "notConfigured"
		DerivedCredentialUrl = $null
		AndroidDeviceAdministratorEnrollmentEnabled = $false
		IgnoreDevicesForUnsupportedSettingsEnabled = $false
		EnableLogCollection = $true
		EnableAutopilotDiagnostics = $true
		EnableEnhancedTroubleshootingExperience = $false
		EnableDeviceGroupMembershipReport = $false
	}
}
Update-MgDeviceManagement -BodyParameter $params`

Or is there a way to import custom scripts?

Thanks in advance!

**Comment by Micke-K** [comment]
Hello!

Thank you for the support!

There is no support for migrating Settings at the moment. This has been asked before and is part of the list. I need to figure out the best way of doing it. My guess is that in most cases, you would not like to import settings but at first setup it would be a good thing. 

Import custom scrips in what sense? Import scripts into Intune or running a custom script during bulk export/import?

Cheers!
